src/components/TopNav/SecondaryNav.tsx

```diff
@@ -40,22 +34,18 @@ export function SecondaryNav({ currentService, activeSection }: SecondaryNavProp
     return () => {
       document.removeEventListener("astro:page-load", handleUrlChange);
     };
-  }, []);
+  }, [serviceLinks]);
 
   const isActiveSectionLink = (linkId: string | undefined) => {
     if (!linkId) return false;
-    return (activeSection || localActiveSection) === linkId;
+    return localActiveSection === linkId;
   };
 
   return (
     <nav className="border-b border-neutral-800 bg-neutral-800 px-4 sm:px-6 py-3 text-sm">
       <div className="flex max-w-6xl mx-auto justify-between items-center">
-        <a href={`/${currentService.toLowerCase()}`}>
-          <img
-            src={currentService === "Kliento" ? klientoLogo.src : ""}
-            alt={`${currentService} logo`}
-            className="h-4 md:h-5 w-auto"
-          />
+        <a href={`/${serviceName.toLowerCase()}`}>
```

Comment:
This is too brittle. That's part of the reason why I suggest to take the whole `NavItem` above, so we won't have to do this.